File: easy/Compressed-Sequence/Compressed-Sequence.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowSequence(line);
        }
    }
    
    static void ShowSequence(string line){
        string[] numStr = line.Split(' ');
        if(line.Length>0){
            if(line.Length==1) Console.WriteLine("1 " + numStr[0]);
            else {
                int counter = 1;
                int index = 0;
                for(int i=1; i<numStr.Length;i++){
                    if(numStr[i].Equals(numStr[index]))
                        counter++;
                    else {
                        Console.Write(counter + " " + numStr[i-1] + " ");
                        counter = 1;
                        index = i;
                    }
                }
                Console.Write(counter + " " + numStr[index]);
                Console.WriteLine();    
            }
        }
    }
}

File: easy/Compressed-Sequence/Compressed-Sequence.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowSequence(line);
        }
    }
    
    static void ShowSequence(String line){
        String[] numStr = line.split("\\s+");
        if(line.length()>0){
            if(line.length()==1) System.out.println("1 " + numStr[0]);
            else {
                int counter = 1;
                int index = 0;
                for(int i=1; i<numStr.length;i++){
                    if(numStr[i].equals(numStr[index]))
                        counter++;
                    else {
                        System.out.print(counter + " " + numStr[i-1] + " ");
                        counter = 1;
                        index = i;
                    }
                }
                System.out.print(counter + " " + numStr[index]);
                System.out.println();    
            }
        }
    }
}
